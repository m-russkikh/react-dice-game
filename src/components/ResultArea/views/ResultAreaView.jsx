import React from 'react';
import { Alert } from 'reactstrap';

import { ResultAreaViewPropType } from '../propTypes';


function ResultAreaView({ win, resultNumber }) {
  if (!resultNumber) {
    return null;
  }

  return (
    <Alert color={win ? 'success' : 'danger'}>
      {resultNumber} {win ? 'WIN' : 'LOSE'}
    </Alert>
  );
}

ResultAreaView.propTypes = ResultAreaViewPropType;

export default ResultAreaView;
