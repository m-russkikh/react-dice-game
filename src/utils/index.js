import { DEFAULT_BALANCE } from '../constants/balance';

export function getPreloadedState() {
  let balance = {
    value: DEFAULT_BALANCE,
  };

  try {
    const serializedBalance = localStorage.getItem('balance');

    if (serializedBalance) {
      balance = JSON.parse(serializedBalance);
    }
  } catch (err) {
  //  ignore errors
  }

  return { balance };
}

export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getNextBetNumber() {
  return Math.floor(Math.random(0, 1) * 100) + 1;
}
