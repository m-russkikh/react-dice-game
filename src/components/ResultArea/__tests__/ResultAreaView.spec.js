import { getShallowWrapper } from '../../../utils/testUtils';
import ResultAreaView from '../views/ResultAreaView';


describe('ResultAreaView', () => {
  describe('render', () => {
    it('should display root element', () => {
      const props = {
        resultNumber: 50,
        win: true,
      };
      const wrapper = getShallowWrapper(ResultAreaView, props);

      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should not display root element', () => {
      const props = {
        resultNumber: null,
        win: true,
      };
      const wrapper = getShallowWrapper(ResultAreaView, props);

      expect(wrapper.find('Alert').exists()).toBeFalsy();
    });
  });
});
