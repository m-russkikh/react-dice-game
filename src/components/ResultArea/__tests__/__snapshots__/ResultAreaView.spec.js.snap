// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultAreaView render should display root element 1`] = `
"<Alert color=\\"success\\" isOpen={true} tag=\\"div\\" closeAriaLabel=\\"Close\\" transition={{...}}>
  50
   
  WIN
</Alert>"
`;
